front/tests/pages/rollingstock-editor-page-model.ts

```diff
@@ -51,6 +75,18 @@ class RollingstockEditorPage extends CommonPage {
     this.powerRestrictionSelector = page.getByTestId('power-restriction-selector');
     this.electricalProfileSelector = page.getByTestId('electrical-profile-selector');
     this.loadingGauge = page.locator('#loadingGauge');
+    this.primaryCategorySelector = page.getByRole('combobox', {
+      name: this.translations.primaryCategory,
+    });
+    this.categoriesCheckboxes = Object.fromEntries(
+      Object.keys(RollingStockCategoryDict).map((category) => [
+        category,
+        page.getByRole('checkbox', {
+          name: this.translations.categoriesOptions[category],
+          exact: true,
+        }),
+      ])
+    ) as Record<RollingStockCategory, Locator>;
```

Comment:
```suggestion
  private getCategoryCheckbox(category: string): Locator {
    return this.page.locator(`#other_category_${category}`);
  }

```
why not use the ID or even add data-testid, it's faster than querying by role and name

Reply:
Playwright recommends using role locators over .locators using id, to be closer to actual user interaction.

> Playwright comes with multiple built-in locators. To make tests resilient, we recommend prioritizing user-facing attributes and explicit contracts such as page.getByRole().

(...)

> CSS and XPath are not recommended as the DOM can often change leading to non resilient tests. Instead, try to come up with a locator that is close to how the user perceives the page such as role locators or define an explicit testing contract using test ids.

Using test ids could also work though

Reply:
Yes, Playwright's built-in locators are recommended, but using `IDs` if they are unique and stable is also a valid choice However, let's go with` page.getByTestId()` since it's a built-in locator and performs a range of actionability checks

Reply:
Sure ^^

Reply:
Added data-testid to primary category selector and category checkboxes, and updated the locators ^^